File: api/src/main/java/gg/mineral/api/inventory/item/enchant/Enchantment.java
package gg.mineral.api.inventory.item.enchant;

import it.unimi.dsi.fastutil.shorts.Short2ObjectOpenHashMap;
import lombok.RequiredArgsConstructor;
import lombok.Value;

@Value
@RequiredArgsConstructor
public class Enchantment {
    short id;
    private final static Short2ObjectOpenHashMap<Enchantment> ENCHANTMENTS = new Short2ObjectOpenHashMap<>();
    // TODO register enchants
    /*
     * static {
     * 
     * }
     */

    public static Enchantment getById(short id) {
        return ENCHANTMENTS.get(id);
    }
}

File: api/src/main/kotlin/gg/mineral/api/inventory/item/enchant/Enchantment.kt
package gg.mineral.api.inventory.item.enchant

import it.unimi.dsi.fastutil.shorts.Short2ObjectOpenHashMap

class Enchantment {
    var id: Short = 0

    companion object {
        private val ENCHANTMENTS = Short2ObjectOpenHashMap<Enchantment>()

        // TODO register enchants
        /*
     * static {
     * 
     * }
     */
        fun getById(id: Short): Enchantment =
            ENCHANTMENTS[id]
    }
}
